### DISCUSSION TITLE
Access `params` in `createServerAction$`?

### DISCUSSION THREAD

**Discussion by binajmen** [solidjs/solid-start#551] [Ideas]
The same way we have access to `request`, it would be nice to have access to `params` as well:
```diff
-const [create, { Form }] = createServerAction$(async (formData: FormData, { request }) => {
+const [create, { Form }] = createServerAction$(async (formData: FormData, { request, params }) => {
    ...
});
```
So we could use `params.foo` with the route `/routes/[foo]/bar`

Don't know if there are reasons I'm unaware of for not doing/being able to do that. Hence the "Ideas" category ;)

**Comment by ryansolid** [comment]
Hmm.. actions don't have a route really that they are associated with. Like in order to send params we'd need to send it as part of the the request. Technically we have the referer I believe so maybe some consideration here but it hasn't been processed by the router so getting it out in such a clean form might be more involved. While I understand this is probably pretty common the suggestion right now is to encode it in a hidden input field in the form, or pass it as a variable to the mutation.

**Reply by binajmen** [reply]
> actions don't have a route really that they are associated with

I was expecting something like that based on one of your live stream. I can live with a hidden field/variable :)
